core/slice.go

```diff
@@ -927,3 +932,20 @@ func (sl *Slice) WriteBlock(block *types.Block) {
 func (sl *Slice) LatestTps() uint32 {
 	return sl.latestTps
 }
+
+func (sl *Slice) AverageAppendTime() int64 {
+	var totalDuration int64
+	for _, key := range sl.appendHistory.Keys() {
```

Comment:
This is not rollback safe and will result in erroneous measurements. Please see [my feedback](https://github.com/dominant-strategies/go-quai/pull/813#discussion_r1178189914) on Connor's PR and follow the same approach:
* append time becomes atomic value in block struct
* append value is always computed as function of parent's append value